auslib/db.py

```diff
@@ -1487,11 +1487,22 @@ def getPotentialRequiredSignoffs(self, affected_rows, transaction=None):
     def _matchesRegex(self, foo, bar):
         # Expand wildcards and use ^/$ to make sure we don't succeed on partial
         # matches. Eg, 3.6* matches 3.6, 3.6.1, 3.6b3, etc.
-        test = foo.replace('.', '\.').replace('*', '.*')
-        test = '^%s$' % test
-        if re.match(test, bar):
-            return True
-        return False
+        if foo.endswith('*'):
+            if(re.match('^.{0,2}\*$', foo)):
+                return False
+            test = foo.replace('.', '\.').replace('*', '\*', foo.count('*')-1)
+            test = test[:-1]
+            test = test + '.*'
+            test = '^%s$' % test
```

Comment:
This section might read a bit better with string construction, eg:
`test = '^{}.*$'.format(test[:-1])`